medmodels/medrecord/medrecord.py

```diff
@@ -1343,130 +1344,83 @@ def clone(self) -> MedRecord:
 
         return medrecord
 
-    def _describe_group_nodes(self) -> pl.DataFrame:
+    def _describe_group_nodes(
+        self,
+    ) -> Dict[Union[Group, Literal["Ungrouped"]], AttributeInfo]:
         """Creates a summary of group nodes and their attributes.
 
         Returns:
             pl.DataFrame: Dataframe with all nodes in medrecord groups and their attributes.
         """
-        df_schema = {
-            "Nodes Group": pl.String,
-            "Count": pl.Int32,
-            "Attribute": pl.String,
-            "Info": pl.String,
-        }
-
-        node_groups = [pl.DataFrame(schema=df_schema)]
+        nodes_info = {}
+        grouped_nodes = []
 
-        groups = set(self.groups)
+        groups = sorted(set(self.groups), key=lambda x: (type(x).__name__, x))
```

Comment:
what does the `key=...` part here do? especially since the key is now a tuple

Reply:
also `self.groups` should already be a set afaik, since groups are unique

Reply:
The key is for the case of str/int mix in the Groups. Without it I got an error when trying to sort the group names. If you have a more elegant way, I'm very interested

Reply:
okay so this sorts by the type name of group first and if there is a sorting collision it sorts by the actual value (could also be described as sorts by type first and then each type is sorted by value).

Why not use `sorted(set(self.groups), key=lambda x: str(x))` - The output should be the same and the sort key is not a tuple (which might be unintended functionality in python and is not documented anywhere I could find except for a stackoverflow post - If you found any documentation on this while implementing this, please let me know).